### ISSUE TITLE
Bug and comment on contrast curves

### ISSUE THREAD

**Issue by majobodo** [vortex-exoplanet/VIP#439]
Hi all,

while reading the code on the contrast curve and comparing it with the math stated in the Mawet et. al 2014 paper if found three bugs in the implementation:

- [ ] Line: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L253

the small sample statistic correction is calculated. In the case of an S/R calculation, this is correct as one resolution element is used by the planet. In the case of the contrast curve, this should be `n_res_ele` without -1.

- [ ] Line: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L254
regarding the scipy documentation the second parameter of ppf, `df` is the degree of freedom. Given e.g. 6 resolution elements we have `df=5`. Thus, we have to subtract `-1` from `n_res_ele`.

- [ ] Line: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L256
and: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L259
and: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L244
and: https://github.com/vortex-exoplanet/VIP/blob/9ca07ade972f8123f7ca2cc03860b0be375fcefa/vip_hci/metrics/contrcurve.py#L246

the term `sigma_corr * noise_samp_sm` can be interpreted as the flux a planet needs to have in order to be detectable under the given confidence level. As stated in Mawet et. al the average of the noise residual elements needs to be added here (x in the notation of the original paper). In other words, the planet has to be brighter if all noise residual elements are slightly brighter than zero.

In addition, a more general comment: Both parameters `fc_snr ` as well as `sigma` are interpreted w.r.t. a Gaussian distribution. In other words, even if `student ` is True the inserted flux of the fake planets is based on the Gaussian distribution. The same applies to sigma. I think this should be stated in the documentation.

Cheers,

Markus

**Comment by VChristiaens** [comment]
Hi Markus,
Thanks a lot for pointing these bugs out. I agree with your interpretation of the Mawet et al. paper and have implemented all your suggestions. For some datasets I had noticed that the contrast curve changed slope very close from the star, hence was likely too optimistic. I believe it may indeed have been due to the missing 'x' term representing the average residual level in the apertures.
Cheers,
Valentin
